src/main/java/brave/opentracing/BraveTracer.java

```diff
@@ -52,7 +54,7 @@
  * @see BraveSpan
  * @see Propagation
  */
-public final class BraveTracer implements Tracer {
+public final class BraveTracer extends ThreadLocalActiveSpanSource implements Tracer {
```

Comment:
One idea to address that problem would be to add hooks in Brave itself to allow the ActiveSpanSource implementation to listen for changes to the CurrentTraceContext. Thoughts?